WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -412,32 +412,32 @@ protected override void OnNavigatedFrom(bool isInHistory)
 
 	private async Task OnConfirmAsync()
 	{
-		var transaction = await Task.Run(() => TransactionHelpers.BuildTransaction(_wallet, _info));
-		var transactionAuthorizationInfo = new TransactionAuthorizationInfo(transaction);
-		var authResult = await AuthorizeAsync(transactionAuthorizationInfo);
-		if (authResult)
+		try
 		{
-			IsBusy = true;
-
-			try
+			var transaction = await Task.Run(() => TransactionHelpers.BuildTransaction(_wallet, _info));
+			var transactionAuthorizationInfo = new TransactionAuthorizationInfo(transaction);
+			var authResult = await AuthorizeAsync(transactionAuthorizationInfo);
+			if (authResult)
 			{
+				IsBusy = true;
+
 				var finalTransaction =
 					await GetFinalTransactionAsync(transactionAuthorizationInfo.Transaction, _info);
 				await SendTransactionAsync(finalTransaction);
 				_wallet.UpdateUsedHdPubKeysLabels(transaction.HdPubKeysWithNewLabels);
 				_cancellationTokenSource?.Cancel();
 				Navigate().To().SendSuccess(_wallet, finalTransaction);
-			}
-			catch (Exception ex)
-			{
-				Logger.LogError(ex);
-				await ShowErrorAsync(
-					"Transaction",
-					ex.ToUserFriendlyString(),
-					"Wasabi was unable to send your transaction.");
-			}
 
-			IsBusy = false;
+				IsBusy = false;
```

Comment:
`IsBusy` is modified inside the try-catch now. Which means in case of an exception `IsBusy` can stay `true`.

Maybe introduce the finally block and move this line there?